galleon-plugins/src/main/java/org/wildfly/galleon/plugin/ShadedModel.java

```diff
@@ -0,0 +1,260 @@
+        }
+        Map<String, String> manifestEntries = getManifestEntries();
+        if (manifestEntries != null) {
+            for (Map.Entry<String, String> entry : manifestEntries.entrySet()) {
+                if (entry.getValue() == null) {
+                    attributes.remove(new Attributes.Name(entry.getKey()));
+                } else {
+                    attributes.put(new Attributes.Name(entry.getKey()), entry.getValue());
+                }
+            }
+        }
+        attributes.put(Attributes.Name.IMPLEMENTATION_TITLE, "Galleon shading of " + getName());
+        attributes.put(Attributes.Name.SPECIFICATION_TITLE, "Galleon shading of " + getName());
+        attributes.put(Attributes.Name.IMPLEMENTATION_VERSION, "Unknown");
+        Files.deleteIfExists(targetManifestPath);
+        try (FileOutputStream out = new FileOutputStream(targetManifestPath.toFile())) {
+            manifest.write(out);
+        }
+        Path moduleInfoPath = target.resolve("module-info.class");
+        Files.deleteIfExists(moduleInfoPath);
+        Path indexListPath = target.resolve("META-INF/INDEX.LIST");
+        Files.deleteIfExists(indexListPath);
+        Path services = target.resolve("META-INF").resolve("services");
+        for (Map.Entry<String, List<String>> entry : serviceLoaders.entrySet()) {
+            Path file = services.resolve(entry.getKey());
+            Files.write(file, entry.getValue(), UTF_8);
+        }
+    }
+
+    @Override
+    public boolean consume(Path resourcePath) throws IOException {
+        String entry = resourcePath.toString().substring(1);
+        if (entry.startsWith("META-INF/MANIFEST.MF")) {
+            if (!seenManifest) {
+                seenManifest = true;
+                return true;
+            } else {
+                // Do not copy other Manifest files
+                return false;
+            }
+        } else if (entry.startsWith("META-INF/services/")) {
+            String fileName = resourcePath.getFileName().toString();
+            List<String> lines = Files.readAllLines(resourcePath);
+            List<String> allLines = serviceLoaders.get(fileName);
+            Set<String> allClasses = classes.get(fileName);
+            if (allLines == null) {
+                allLines = new ArrayList<>();
+                serviceLoaders.put(fileName, allLines);
+            }
+            if (allClasses == null) {
+                allClasses = new HashSet<>();
+                classes.put(fileName, allClasses);
+            }
+            boolean newClasses = false;
+            for (String l : lines) {
+                if (!l.trim().startsWith("#")) {
+                    if (!allClasses.contains(l)) {
+                        newClasses = true;
+                        break;
+                    }
+                }
+            }
+            if (newClasses) {
+                for (String l : lines) {
+                    if (l.trim().startsWith("#")) {
```

Comment:
Should this condition be negated like on line 235? I assume `#` marks a comment - or is it something different?